src/test/java/uk/gov/pay/directdebit/mandate/dao/MandateSearchDaoIT.java

```diff
@@ -191,4 +184,25 @@ public void searchByMultipleParams() {
         var total = mandateSearchDao.countTotalMatchingMandates(searchParams, gatewayAccountFixture.getExternalId());
         assertThat(total).isEqualTo(1);
     }
+
+    @Test
+    public void searchByExternalState() {
+        var mandateWithWrongState = aMandateFixture()
+                .withGatewayAccountFixture(gatewayAccountFixture)
+                .withId(999L)
+                .withState(MandateState.EXPIRED)
+                .insert(testContext.getJdbi());
+
+        var searchParams = aMandateSearchParams()
+                .withExternalMandateState("pending")
+                .build();
+
+        var results = mandateSearchDao.search(searchParams, gatewayAccountFixture.getExternalId());
+        assertThat(results).hasSize(2);
+        assertThat(results.get(0).getState().toExternal().getState()).isEqualTo("pending");
+        assertThat(results.get(1).getState().toExternal().getState()).isEqualTo("pending");
```

Comment:
Might be easier just to assert that we specifically get `mandate1` and `mandate2` — I think it would make the intent a bit clearer

Reply:
yes that is clearer!